docs/getting-started/application/patterns.rst

```diff
@@ -53,122 +53,61 @@ The above policy controls access to the Expense model on the **record level**.
-is evaluated.
-
-.. _second-record-level:
-
-.. code-block:: python
-    :caption: :fab:`python`
-
-    def get_expense(user, expense_id):
-        expense = db.fetch(
-            "SELECT * FROM expenses WHERE id = %", expense_id)
-
-        if oso.is_allowed(user, "view", expense):
-            # Process request
-            ...
-        else:
-            # Not authorized
-            return NotAuthorizedResponse()
-
-This **policy evaluation point** is **after primary data fetch**. An
-authorization decision is made after the primary data is fetched from the
-persistence layer (be it a SQL database, an ORM, or another service) and can be
-used to make an authorization decisions.
-
-Alternatively, we could have placed our policy evaluation point **before primary
-data fetch**. This would limit the power of our policy, since we would not be
-able to check the ``amount`` field of the ``Expense``. Keep reading to see how
-we would apply this technique to **model level** and **field level** access
-control.
-
-.. tip::
-
-    A best practice for this type of access control is to integrate the policy
-    evaluation point within the **data access layer**.  This
-    ensures that the ``oso.is_allowed`` call is made when an expense is accessed, no
-    matter where that access occurs in the application.
-
-.. todo::
-
-   Would be nice to be able to say here: "see this guide for an
-   example of how to do this..."
-
 Model Level Access Control
 ==========================
 
-Sometimes, access control does not rely on properties of the primary data.  This
-type of access control is called **model level**.
+Sometimes, access control does not depend on the properties of the resource but just
+on it's type. This is called **model level** can be used for coarse level authorization.
+This is useful if you want to restrict access to a whole page if the user cannot see any
+of the resources of the kind the page is about. Since the rules depend only on the type
+of the model and not on the data it makes it easier to add these checks upstream in
+middleware.
 
 .. code-block:: polar
     :caption: :fa:`oso`
 
-    allow(actor, "view", "expense") if actor.role = "accountant";
-    allow(actor, "modify", "team") if actor.role = "hr_admin";
-    allow(actor, "modify", "project") if actor.role = "hr_admin";
-    allow(actor, "modify", "organization") if actor.role = "hr_admin";
+    allow(actor, "view", "Expense") if actor.role = "accountant";
+    allow(actor, "modify", "Team") if actor.role = "hr_admin";
+    allow(actor, "modify", "Project") if actor.role = "hr_admin";
+    allow(actor, "modify", "Organization") if actor.role = "hr_admin";
 
 This brief policy shows an example of model level access control:
 
 - An accountant can view expenses.
 - HR admins can modify teams, projects, and organizations.
 
-Notice that this policy does not rely on any **primary authorization data**.
-Therefore it can be evaluated either before or after the primary data fetch.
```

Comment:
Think it might be worth keeping this in to reinforce the point, but replace "primary authorization data" with "resource attributes" or something like that.